exchanges/gateio/gateio_wrapper.go

```diff
@@ -899,39 +900,26 @@ func (g *Gateio) FormatExchangeKlineInterval(in kline.Interval) string {
 }
 
 // GetHistoricCandles returns candles between a time period for a set time interval
-func (g *Gateio) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
-	if err := g.ValidateKline(pair, a, interval); err != nil {
-		return kline.Item{}, err
-	}
-
-	hours := time.Since(start).Hours()
-	formattedPair, err := g.FormatExchangeCurrency(pair, a)
+func (g *Gateio) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, required kline.Interval, start, end time.Time) (*kline.Item, error) {
+	builder, err := g.GetKlineBuilder(pair, a, required, start, end)
 	if err != nil {
-		return kline.Item{}, err
-	}
-
-	params := KlinesRequestParams{
-		Symbol:   formattedPair.String(),
-		GroupSec: g.FormatExchangeKlineInterval(interval),
-		HourSize: int(hours),
+		return nil, err
 	}
 
-	klineData, err := g.GetSpotKline(ctx, params)
+	klineData, err := g.GetSpotKline(ctx, KlinesRequestParams{
+		Symbol:   builder.Formatted.String(),
+		GroupSec: g.FormatExchangeKlineInterval(builder.Request),
+		HourSize: int(time.Since(builder.Start).Hours()),
+	})
 	if err != nil {
-		return kline.Item{}, err
+		return nil, err
 	}
-	klineData.Interval = interval
-	klineData.Pair = pair
-	klineData.Asset = a
-
-	klineData.SortCandlesByTimestamp(false)
-	klineData.RemoveOutsideRange(start, end)
-	return klineData, nil
+	return builder.ConvertCandles(klineData)
 }
 
 // GetHistoricCandlesExtended returns candles between a time period for a set time interval
-func (g *Gateio) GetHistoricCandlesExtended(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
-	return g.GetHistoricCandles(ctx, pair, a, start, end, interval)
+func (g *Gateio) GetHistoricCandlesExtended(_ context.Context, _ currency.Pair, _ asset.Item, _ kline.Interval, _, _ time.Time) (*kline.Item, error) {
+	return nil, common.ErrNotYetImplemented
```

Comment:
It's basically a burden of proof shift where the candle request limit would be called by the client and that would dictate which function to utilize and this function is for walking through historic blocks which will require more than one request. This is to limit complexity at the wrapper level. I will be going further on this down to the builder level in later PR's to reject instead of repair, sort, filter approach.

Reply:
I will add a more description to PR header as to what future PR's will entail.